src/datamodel/ExpectedElement.cpp

```diff
@@ -8,5 +8,6 @@ ExpectedElement::ExpectedElement(std::string name, std::string ns)
 
 const char* ExpectedElement::what() const noexcept
 {
-  return "expected element not encountered";
+  auto msg = "expected element " + name_ + " from namespace " + ns_ + " not encountered";
+  return msg.c_str();
```

Comment:
This causes undefined behaviour! You are returning a pointer to stack memory associated with a local variable! C++ standard states the pointer has to be valid at least until the exception object is destroyed or a non-const member function is called!